### PR TITLE
use go's built-in runtime arch

### PR DESCRIPTION
**Pull request by clnperez**
If the user hasn't explictyly set GOARCH, don't default to amd64. This
way, a user has to expliciltly cross-compile, instead of needing to know
to set GOARCH on build for a native arch that isn't amd64.

Signed-off-by: Christy Norman <christy@linux.vnet.ibm.com>

**Description of the change:**

See https://github.com/operator-framework/operator-sdk/pull/2256

**Motivation for the change:**

See https://github.com/operator-framework/operator-sdk/pull/2256

### CODE REVIEW

File: cmd/operator-sdk/build/cmd.go [hunk 1]
```diff
@@ -93,8 +93,6 @@ func buildFunc(cmd *cobra.Command, args []string) error {
 
 	if value, ok := os.LookupEnv("GOARCH"); ok {
 		goBuildEnv = append(goBuildEnv, "GOARCH="+value)
-	} else {
-		goBuildEnv = append(goBuildEnv, "GOARCH=amd64")
 	}
```
Comment (joelanford) [inline cmd/operator-sdk/build/cmd.go]: This entire section can be removed. `goBuildEnv` is setup from the current environment int line 92, so line 95 doesn't actually do anything. Either GOARCH is already set in the environment, or it is unset, in which case go will use the default GOARCH (based on the architecture of the `go` binary itself).

Comment (joelanford) [reply cmd/operator-sdk/build/cmd.go]: Also probably need a CHANGELOG update for this change.

I'm thinking a bugfix:
```
Updates `operator-sdk build` for go operators to compile the operator binary based on Go's built-in GOARCH detection. This fixes an issue that caused an `amd64` binary to be built into non-`amd64` base images when using operator-sdk on non-`amd64` architectures. ([#2268](https://github.com/operator-framework/operator-sdk/pull/2268))
```

Comment (clnperez) [reply cmd/operator-sdk/build/cmd.go]: Updated and tested @joelanford

**Comment by camilamacedo86** [comment]
Merging this one 👍

**Comment by clnperez** [comment]
Sorry - was out on vacation. Thanks all!

**Pull request merged**